analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -34,9 +34,11 @@ public abstract class SonarAnalysisContextBase
 
     public abstract bool TryGetValue<TValue>(SourceText text, SourceTextValueProvider<TValue> valueProvider, out TValue value);
 
-    public bool ShouldAnalyze(GeneratedCodeRecognizer generatedCodeRecognizer, SyntaxTree tree, Compilation compilation, AnalyzerOptions options) =>    // FIXME: This thing has confusing name
-        !IsUnchanged(tree, compilation, options)        // FIXME: This needs to go to
-        && (ShouldAnalyzeGenerated(compilation, options) || !tree.IsGenerated(generatedCodeRecognizer, compilation));
+    /// <param name="tree">Tree to decide on. Can be null for Symbol-based and Compilation-based scenarios. And we want to analyze those too.</param>
+    /// <param name="generatedCodeRecognizer">When set, generated trees are analyzed only when langauge-specific 'analyzeGeneratedCode' configuration property is also set.</param>
+    public bool ShouldAnalyzeTree(SyntaxTree tree, Compilation compilation, AnalyzerOptions options, GeneratedCodeRecognizer generatedCodeRecognizer) =>
```

Comment:
The method name should be `ShouldAnalyze`. I find `Tree` redundant. We already provide the syntax tree as a parameter.

Reply:
The method was originally called `ShouldAnalyze` and it confused us during incremental PR analysis. I like the `Tree` suffix here to highlight that fact. While the tree is first argument, the method also receives `Compilation` and `AnalyzerOptions` and it's not clear that they are secondary noise.